src/v/transform/rpc/serde.cc

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "transform/rpc/serde.h"
+
+#include "model/record.h"
+
+#include <seastar/core/chunked_fifo.hh>
+
+namespace transform::rpc {
+transformed_topic_data::transformed_topic_data(
+  model::topic_partition tp, model::record_batch b)
+  : tp(std::move(tp)) {
+    ss::chunked_fifo<model::record_batch> batches;
+    batches.reserve(1);
+    batches.push_back(std::move(b));
+}
+transformed_topic_data::transformed_topic_data(
+  model::topic_partition tp, ss::chunked_fifo<model::record_batch> b)
+  : tp(std::move(tp))
+  , batches(std::move(b)) {}
+
+transformed_topic_data transformed_topic_data::share() {
+    ss::chunked_fifo<model::record_batch> shared;
+    shared.reserve(batches.size());
+    for (auto& batch : batches) {
+        shared.push_back(batch.share());
+    }
+    return {tp, std::move(shared)};
+}
```

Comment:
question: Maybe this is required by serde, but why not use a copy constructor or a copy assignment operator

Reply:
The copy constructor/assignment is disabled for batches. And this shares the underlying iobuf (no copies), it can be useful if you want to keep an iobuf :)

Reply:
For example if you wanted to do retries, you'd just share a batch each time and the underlying iobuf would be ref counted and not copied as far as I understand it.